### ISSUE TITLE
Node Graph: Duplicate data links

### ISSUE THREAD

**Issue by gelicia** [grafana/grafana#96850]
### What happened?

Using the Node Graph scenario on the test data source with the Node Graph visualization, then creating a test data link, shows the data link multiple times.

![Image](https://github.com/user-attachments/assets/8396e5a8-03ae-45e0-b778-c0c20d16b346)

### What did you expect to happen?

The link to show once.

### Did this work before?

Not sure.

### How do we reproduce it?

1. Create a node graph panel using the node graph scenario of the test data source.
2. Add a data link
3. Click a node to see the duplicate links.

### Is the bug inside a dashboard panel?

<details><summary>Example Dashboard</summary>
This uses a test data datasource.

```json
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": {
          "type": "grafana",
          "uid": "-- Grafana --"
        },
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": 232,
  "links": [],
  "panels": [
    {
      "datasource": {
        "type": "grafana-testdata-datasource",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "links": [
            {
              "title": "Test",
              "url": "google.com"
            }
          ]
        },
        "overrides": []
      },
      "gridPos": {
        "h": 8,
        "w": 12,
        "x": 0,
        "y": 0
      },
      "id": 1,
      "options": {
        "edges": {},
        "nodes": {},
        "zoomMode": "cooperative"
      },
      "pluginVersion": "11.4.0-pre",
      "targets": [
        {
          "refId": "A",
          "scenarioId": "node_graph"
        }
      ],
      "title": "Panel Title",
      "type": "nodeGraph"
    }
  ],
  "preload": false,
  "schemaVersion": 40,
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {},
  "timezone": "browser",
  "title": "Node Graph",
  "uid": "be4mnqmd3sr9cf",
  "version": 2,
  "weekStart": ""
}
```
</details>

### Environment (with versions)?

Grafana: main

### Grafana platform?

None

### Datasource(s)?

Test Datasource

**Comment by gelicia** [comment]
> fwiw I'm getting multiple datalinks for every edge/node I click on. Just using a really simple local test setup so seems like something is wrong.
> 
> ![Image](https://github.com/user-attachments/assets/c029c4c9-234f-4a7e-b589-71e2cc1494e3)

Just wanted to confirm @jkemp101  - this is not using the test datasource to render that node graph?

**Comment by jkemp101** [comment]
Nope. Just a local sqlite db I'm playing with. I deleted the authalligator node row to see if the number of links changed, other than that it looks like this:

![Image](https://github.com/user-attachments/assets/29872684-2e0b-4c42-82b9-14ad47f0684e)
![Image](https://github.com/user-attachments/assets/2e99b0b6-9321-44df-9723-8d1b9c6193cb)

**Comment by Willing-Xyz** [comment]
I resolved the issue by using field override.
![Image](https://github.com/user-attachments/assets/639fae3c-3311-4e91-a0bc-953d6b4861b4)

**Comment by andalibmalit** [comment]
Running into the same issue even when using a field override. The generated data links (creates two duplicates) all work, but do not respect unchecking "Open in new tab". Querying a local InfluxDB database

![Image](https://github.com/user-attachments/assets/369347db-13c8-4093-8add-f00d4645fe11)
![Image](https://github.com/user-attachments/assets/60e7e921-9810-454b-871c-0b4e0929a7e6)
